### ISSUE TITLE
Reject unsupported admin usernames for Postgres servers

### ISSUE THREAD

**Issue by whiskeysierra** [terraform-providers/terraform-provider-azurerm#10726]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The login name for the postgres administrators should be validated.

> Enter your server admin username. It can't start with pg_ and these values are not allowed: azure_superuser, azure_pg_admin, admin, administrator, root, guest, or public.
>
> https://docs.microsoft.com/en-us/azure/postgresql/quickstart-create-server-database-portal

### New or Affected Resource(s)

* azurerm_postgresql_server (administrator_login)
* azurerm_postgresql_active_directory_administrator (login)

### Potential Terraform Configuration

```hcl
resource "azurerm_postgresql_active_directory_administrator" "example" {
  server_name         = azurerm_postgresql_server.example.name
  resource_group_name = azurerm_resource_group.example.name
  login               = "admin" # won't work
  tenant_id           = data.azurerm_client_config.current.tenant_id
  object_id           = data.azurerm_client_config.current.object_id
}
```

### References

* https://docs.microsoft.com/en-us/azure/postgresql/quickstart-create-server-database-portal

**Comment by HarleyB123** [comment]
Hey 👋

PR #10757 fixes this  😅

**Comment by tombuildsstuff** [comment]
Fixed via #10757, thanks @HarleyB123 :)
